dff/context_storages/mongo.py

```diff
@@ -11,84 +11,195 @@
+import time
+from typing import Hashable, Dict, Union, Optional, List, Any
 
 try:
     from motor.motor_asyncio import AsyncIOMotorClient
-    from bson.objectid import ObjectId
 
     mongo_available = True
 except ImportError:
     mongo_available = False
     AsyncIOMotorClient = None
-    ObjectId = Any
-
-import json
+    ObjectId = None
 
 from dff.script import Context
 
-from .database import DBContextStorage, threadsafe_method
+from .database import DBContextStorage, threadsafe_method, cast_key_to_string
 from .protocol import get_protocol_install_suggestion
+from .context_schema import ALL_ITEMS, FieldDescriptor, ValueSchemaField, ExtraFields
 
 
 class MongoContextStorage(DBContextStorage):
     """
     Implements :py:class:`.DBContextStorage` with `mongodb` as the database backend.
 
+    Context value fields are stored in `COLLECTION_PREFIX_contexts` collection as dictionaries.
+    Extra field `_id` contains mongo-specific unique identifier.
+
+    Context dictionary fields are stored in `COLLECTION_PREFIX_FIELD` collection as dictionaries.
+    Extra field `_id` contains mongo-specific unique identifier.
+    Extra fields starting with `__mongo_misc_key` contain additional information for statistics and should be ignored.
+    Additional information includes primary identifier, creation and update date and time.
+
     :param path: Database URI. Example: `mongodb://user:password@host:port/dbname`.
     :param collection: Name of the collection to store the data in.
     """
 
-    def __init__(self, path: str, collection: str = "context_collection"):
+    _CONTEXTS = "contexts"
+    _MISC_KEY = "__mongo_misc_key"
+    _ID_KEY = "_id"
+
+    def __init__(self, path: str, collection_prefix: str = "dff_collection"):
         DBContextStorage.__init__(self, path)
         if not mongo_available:
             install_suggestion = get_protocol_install_suggestion("mongodb")
             raise ImportError("`mongodb` package is missing.\n" + install_suggestion)
-        self._mongo = AsyncIOMotorClient(self.full_path)
+        self._mongo = AsyncIOMotorClient(self.full_path, uuidRepresentation="standard")
         db = self._mongo.get_default_database()
-        self.collection = db[collection]
 
-    @staticmethod
-    def _adjust_key(key: Hashable) -> Dict[str, ObjectId]:
-        """Convert a n-digit context id to a 24-digit mongo id"""
-        new_key = hex(int.from_bytes(str.encode(str(key)), "big", signed=False))[3:]
-        new_key = (new_key * (24 // len(new_key) + 1))[:24]
-        assert len(new_key) == 24
-        return {"_id": ObjectId(new_key)}
+        self.seq_fields = [
+            field
+            for field, field_props in dict(self.context_schema).items()
+            if not isinstance(field_props, ValueSchemaField)
+        ]
+        self.collections = {field: db[f"{collection_prefix}_{field}"] for field in self.seq_fields}
+        self.collections.update({self._CONTEXTS: db[f"{collection_prefix}_contexts"]})
 
```

Comment:
Possible solution for indexation
```suggestion
        self.create_indexes()

    def create_indexes(self):
        primary_id_key = f"{self._MISC_KEY}_{ExtraFields.primary_id}"
        asyncio.run(asyncio.gather(
            self.collections[self._CONTEXTS].create_index([(ExtraFields.primary_id, ASCENDING)], background=True),
            self.collections[self._CONTEXTS].create_index([(ExtraFields.storage_key, HASHED)], background=True),
            self.collections[self._CONTEXTS].create_index([(ExtraFields.active_ctx, HASHED)], background=True),
            *[
                value.create_index([(primary_id_key, ASCENDING)], background=True, unique=True)
                for key, value in self.collections.items()
                if key != self._CONTEXTS
            ]
        ))
```